pkg/render/egressgateway/egressgateway.go

```diff
@@ -287,6 +290,18 @@ func (c *component) egwInitEnvVars() []corev1.EnvVar {
 	}
 }
 
+func (c *component) egwPullSecrets() []*corev1.Secret {
+	var secrets []*corev1.Secret
+	for _, secret := range c.config.PullSecrets {
+		x := secret.DeepCopy()
+		x.ObjectMeta = metav1.ObjectMeta{Name: secret.Name, Namespace: c.config.EgressGW.Namespace}
+		x.ObjectMeta.Labels = common.MapExistsOrInitialize(x.ObjectMeta.Labels)
+		x.ObjectMeta.Labels[common.MultipleOwnersLabel] = "true"
```

Comment:
We should have a comment here explaining why we put the multiple owners label on.

Something like this:

> Each pull secret is shared across all of the EGW deployments in this namespace. As such, we mark it as having multiple owners so that we maintain multiple owner references when creating the secret so that it will only be GC'd when all of its owners have been deleted.